buildSrc/src/main/java/GVNightlyVersionVerifier.kt

```diff
@@ -0,0 +1,210 @@
+        private const val REPO_OWNER = "mozilla-mobile"
+        private const val REPO_NAME = "android-components"
+        private const val BASE_BRANCH_NAME = "master"
+        private const val HEAD = "MickeyMoz"
+    }
+
+    override fun apply(project: Project) {
+
+        project.task("verifyGVNightlyVersion") {
+            setVariables()
+
+            dependsOn("dependencyUpdates")
+
+            doLast {
+                verifyNewGVNightlyVersion(project)
+            }
+        }
+    }
+
+    private fun verifyNewGVNightlyVersion(project: Project) {
+
+        val newVersion = getLastGeckoViewNightlyVersion(project)
+
+        if (newVersion.isNotEmpty()) {
+
+            updateConfigFileWithNewGVNightlyVersion(newVersion)
+
+            val token = project.property("token").toString()
+            val client = GitHubClient(token)
+
+            if (runTests()) {
+
+                if (openAPROnGitHub(client, newVersion))
+                    println("PR Opened Successfully")
+                else
+                    println("Error Creating PR")
+
+            } else {
+                println("Tests failed creating Github Issue")
+                createIssue(client, newVersion)
+            }
+        } else {
+            println("Bad luck :/ no new version of GeckoViewNightly available")
+        }
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    private fun getLastGeckoViewNightlyVersion(project: Project): String {
+        val file = File(project.rootDir, "build" + File.separator +
+                "dependencyUpdates" + File.separator + "report.json")
+
+        val json = JsonSlurper().parse(file)
+
+        val outdated = (json as LazyMap)["outdated"]
+        val dependencies = (outdated as Map<*, *>)["dependencies"] as ArrayList<LazyMap>
+
+        val gecko = dependencies.filter {
+            (it["name"] as String).contains("geckoview-nightly")
+        }
+
+        val lastVersion =
+                if (gecko.isNotEmpty())
+                    ((gecko.first()["available"] as LazyMap)["milestone"]) as String
+                else ""
+
+        val actualVersionMajor = nightly_version.split(".")
+        val lastVersionMajor = lastVersion.split(".")
+
+        return if (actualVersionMajor == lastVersionMajor) lastVersion else ""
+    }
+
+    private fun setVariables() {
+        System.setProperty("outputFormatter", "json")
+        System.setProperty("Drevision", "release")
+    }
+
+    private fun updateConfigFileWithNewGVNightlyVersion(newVersion: String) {
+        val file = File(GV_VERSION_PATH_FILE)
+        var fileContent = file.readText()
+        fileContent = fileContent.replace(Regex("nightly_version.*=.*"),
+                "nightly_version = \"$newVersion\"")
+        file.writeText(fileContent)
+        println("${file.name} file updated")
+    }
+
+    private fun runTests(): Boolean {
+        println("Running tests")
+
+        val result = "./gradlew clean test".runCommand()
```

Comment:
It feels a bit weird to run the gradle command from inside a gradle plugin.

I wonder if we could split this into separate tasks:
* update geckoview dependency (just does the update of the file)
* open the PR

The advantage would be:
* I can run the update script locally without opening the PR etc.
* In automation we can run: 
  ```bash
       ./gradlew ....updateGeckoView...    (needs to fail if it couldn't update)
       && ./gradlew clean assemble test 
       && ./gradlew ....openPR...
   ```
* We can use the "open pr" task for other things in the future :)

Reply:
Yes, It felt like the inception movie :P
Nice :)

Reply:
Done :) !

Now there are two plugins, one for just updating the GV version and other for opening pull requests and issues, I updated the PR description with more info.

`./gradlew updateGVNightlyVersion && ./gradlew clean assemble test && git checkout -B newBranchName && git add buildSrc/src/main/java/Gecko.kt && git commit -m "Updating Gecko View Nightly Version" && git push -u -f origin newBranchName && ./gradlew openPR -Ptitle="GV Nightly version Updated" -Pbranch="newBranchName"`